### PR TITLE
Add Product models and API

### PR DESCRIPTION
**Pull request by jerivas**
Fields based on the mockups from the company journey

Additionally I've added changes to configure flake8 as linter and black as formatter, and versioned the API endpoint.

### CODE REVIEW

File: backend/catalogue/models.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+class ProductCategory(models.Model):
+    """
+    Useful to filter products by their features
+    """
+
+    name = models.CharField(_("name"), max_length=100)
+
+    class Meta:
+        verbose_name = _("product category")
```
Comment (vellip) [inline backend/catalogue/models.py]: Isn't that unnecessary because that's the default?

File: backend/catalogue/models.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    category = models.ForeignKey(
+        ProductCategory, verbose_name=_("Category"), on_delete=models.PROTECT
+    )
+    company = models.ForeignKey(
+        "companies.Company", verbose_name=_("Company"), on_delete=models.CASCADE
+    )
+
+    class Meta:
+        verbose_name = _("Product")
+        verbose_name_plural = _("Products")
```
Comment (vellip) [inline backend/catalogue/models.py]: These are also unnecessary because default.

File: backend/catalogue/models.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+class ProductCategory(models.Model):
+    """
+    Useful to filter products by their features
+    """
+
+    name = models.CharField(_("name"), max_length=100, unique=True)
+
+    class Meta:
+        verbose_name = _("product category")
+        verbose_name_plural = _("product categories")
```
Comment (vellip) [inline backend/catalogue/models.py]: Can you title-case this, please?

File: backend/catalogue/models.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    """
+
+    created = models.DateTimeField(_("Created"), auto_now_add=True)
+    updated = models.DateTimeField(_("Last updated"), auto_now=True)
+
+    name = models.CharField(_("Name"), max_length=100)
+    image = models.FileField(_("Image"), upload_to="product-images", max_length=500)
+    price = models.DecimalField(_("Price"), max_digits=8, decimal_places=2)
+    stock = models.PositiveIntegerField(_("In stock"), default=1)
+    description = models.TextField(_("Description"))
```
Comment (vellip) [inline backend/catalogue/models.py]: Should maybe be 
```suggestion
    description = models.TextField(_("Description"), blank=True)
```

File: .gitignore [hunk 5]
```diff
@@ -7,6 +7,7 @@
 /mediafiles/*
 __pycache__/
 /.vscode/
+/product-images
```
[the remark this answers is not in the document]
Comment (jerivas) [reply .gitignore]: Yep, that was a mistake on .gitignore 😅

**Comment by jerivas** [comment]
Hey, sorry for the delay

I couldn't continue with the changes, but glad you could merge it in and apply the fixes 👍🏻

**Comment by vellip** [comment]
No worries! :)

**Pull request merged**